themes/digital.gov/layouts/shortcodes/button.html

```diff
@@ -1,15 +1,3 @@
-{{- $thisHref := .Get "href" -}}
-{{- $thisHref := replaceRE "^/" "" $thisHref -}}
-{{- $.Scratch.Set "thisHref" $thisHref -}}
-{{- $prefix := replaceRE "^(http[s]://).+$" "$1" $thisHref -}}
-{{- $suffix := replaceRE "^.+(.{3,3})$" "$1" $thisHref -}}
-
-{{- if ne $prefix $thisHref -}}
-{{- else if eq $suffix ".md" -}}
-  {{- $.Scratch.Set "thisHref" (relref $.Page $thisHref) -}}
-{{- else -}}
-    {{- $.Scratch.Set "thisHref" ($thisHref | absURL) -}}
-{{- end -}}
-{{- $thisHref := $.Scratch.Get "thisHref" -}}
-
-<a class="usa-button usa-button--outline" href="{{- $thisHref -}}">{{- .Get "text" -}}</a>
+{{ if .Params }}
+  {{ partial "core/usa-button.html" (dict "href" .Params.href "text" .Params.text) }}
+{{ end }}
```

Comment:
An optional approach is to pass context via `.`, but then you have to have additional checks in component to differentiate if it was called via shortcode or partial.

Chose this approach for now to not complicate logic in component further.

Reply:
Hugo docs recommend this approach, looks good.